packages/contracts-bedrock/contracts/libraries/Encoding.sol

```diff
@@ -10,106 +10,114 @@ import { RLPWriter } from "./rlp/RLPWriter.sol";
  */
 library Encoding {
     /**
-     * Generates the correct cross domain calldata for a message.
-     * @param _target Target contract address.
-     * @param _sender Message sender address.
-     * @param _message Message to send to the target.
-     * @param _messageNonce Nonce for the provided message.
-     * @return ABI encoded cross domain calldata.
+     * @notice RLP encodes the L2 transaction that would be generated when a given deposit is sent
+     *         to the L2 system. Useful for searching for a deposit in the L2 system.
+     *
+     * @param _from        Address of the sender of the deposit.
+     * @param _to          Address of the receiver of the deposit.
+     * @param _value       ETH value to send to the receiver.
+     * @param _mint        ETH value to mint on L2.
+     * @param _gasLimit    Gas limit to use for the transaction.
+     * @param _isCreation  Whether or not the transaction is a contract creation.
+     * @param _data        Data to send with the transaction.
```

Comment:
Not directly related to this PR, but these values get passed around together all the time. WDYT on putting them into a struct and treating them as a proper type?

Reply:
Strongly agree it should be a type